extensions/amp-story/1.0/amp-story.js

```diff
@@ -2322,6 +2320,33 @@ export class AmpStory extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Loads and initializes the amp-story-interactive components if any.
+   * @private
+   */
+  maybeInitializeInteractives_() {
+    const interactives = this.element.querySelectorAll(
+      'amp-story-interactive-quiz, amp-story-interactive-binary-poll, amp-story-interactive-poll'
+    );
+    if (interactives.length > 0) {
+      Services.extensionsFor(this.win)
+        .installExtensionForDoc(
+          this.getAmpDoc(),
+          'amp-story-interactives',
+          '1.0'
+        )
```

Comment:
What's our recommendation for whether we want publishers to add the script to load the extension, or lazy load it ourselves?
I like the fallback, but we'll have to make a decision when we write the validation. Can you do some performance benchmarking with various network/cpu capabilities and interactive components on first/future pages so we can come up with ideal recommendations?
cc @newmuis FYI

Reply:
I'm not sure it's a huge lift for folks to add the script import, and I would expect it to be a performance win, even if it's a small one.  Can we say:

- if the functionality is a publisher choice, require script imports
- if the functionality is a player choice (e.g. through open source web player, or viewer `cap` params), load it ourselves?

The only case that might violate this is of we think we really want to invest in managing the initial load ourselves, to the point that we would want to avoid downloading the binary until later (with the trade-off being that it probably would delay the initial download in the case that we do want it early)

Reply:
> The only case that might violate this is of we think we really want to invest in managing the initial load ourselves

That was my point yes :)) It'd be nice to know the performance difference between both approaches, to see if they're almost equivalent or if one improves/degrades performance so much that it drives our decision.

Reply:
It's safe to say that lazy loading the script is perfect if most of the interactives are not on the first page. Also, it would have similar results as the amp-story-education extension that is also lazy-loaded. Do we have any benchmarks on that?

With some pointers on how to do it, I could benchmark both approaches for this extension and get the data to decide.
Locally it loads instantly, and I assume if the js files get cached, it would also be instant for good-performing devices.

Reply:
We'll require the script on the creator side in the end to avoid any layout shifts on slow connections